packages/backend/src/tools/uif/ManagedChildIndexer.ts

```diff
@@ -31,11 +32,36 @@ export abstract class ManagedChildIndexer extends ChildIndexer {
     assetUniqueIndexerId(this.indexerId)
   }
 
-  async initialize() {
-    const safeHeight = await this.options.indexerService.getSafeHeight(
+  async initialize(): Promise<{ safeHeight: number; configHash?: string }> {
+    const indexerState = await this.options.indexerService.getIndexerState(
       this.indexerId,
     )
-    return safeHeight ?? this.options.minHeight - 1
+
+    if (
+      this.options.configHash &&
+      this.options.configHash !== indexerState?.configHash
+    ) {
+      return {
+        safeHeight: this.options.minHeight - 1,
+        configHash: this.options.configHash,
+      }
+    }
+
+    return {
+      safeHeight: indexerState?.safeHeight ?? this.options.minHeight - 1,
```

Comment:
You can add `indexerState === undefined` as a constraint in the previous if statement, but it's a nit pick